### PR TITLE
update python client with latest spec

### PR DESCRIPTION
**Pull request by Rijul1204**
- Updated the client with the latest spec which introduced the concept of a separate user per cluster
- Updated the client to use our newly revamped image embedding API.

### CODE REVIEW

File: python/bagel/api/Cluster.py [hunk 1]
```diff
@@ -42,13 +42,13 @@ class Cluster(BaseModel):
...
 
     def __init__(
-        self,
-        client: "API",
-        name: str,
-        id: UUID,
-        cluster_size: float,
-        embedding_size: Optional[int] = None,
-        metadata: Optional[ClusterMetadata] = None,
+            self,
```
[the remark this answers is not in the document]
Comment (Rijul1204) [reply python/bagel/api/Cluster.py:45]: I use the default IntelliJ formatting whenever I change a file. These indentation changes probably are not congruent with all the other files of this repo. 
But, I can remove these indentation changes if you want. Let me know your thoughts.

[the remark this answers is not in the document]
Comment (BidhanRoy) [reply python/bagel/api/Cluster.py:45]: yes, please keep all the indentation formats in all files same. even better if you setup for linter on the repo for future mismatch. this is an open source repo, will be useful open source contributors as well.

[the remark this answers is not in the document]
Comment (Rijul1204) [reply python/bagel/api/Cluster.py:45]: Ok, sure. Will remove the indentation in this PR then.
I'll work on setting up the linter for this repo too separately.

[the remark this answers is not in the document]
Comment (Rijul1204) [reply python/bagel/api/Cluster.py]: Pushed a new commit to address review comments

**Review by BidhanRoy — APPROVED** [review]

**Pull request merged**